opencart 3.0.0.0 - 3.0.3.1/upload/admin/controller/extension/module/smaily_for_opencart.php

```diff
@@ -120,7 +146,7 @@ public function index() {
         $data['sync_token_title'] = $this->language->get('sync_token_title');
         $data['sync_token_placeholder'] = $this->language->get('sync_token_placeholder');
         $data['sync_customer_url_title'] = $this->language->get('sync_customer_url_title');
-        $data['customer_cron_url'] = $this->config->get('config_url') . 'index.php?route=extension/smailyforopencart/cron_customers&token=[token]';
+        $data['customer_cron_url'] = $url->link('extension/smailyforopencart/cron_customers') . "&token=" . $data['sync_token'];
```

Comment:
You can use `Url::link` method to generate the required URL without string concatenation:
```suggestion
        $data['customer_cron_url'] = $url->link('extension/smailyforopencart/cron_customers', array('token' => $data['sync_token']));
```

And you should provide 3rd argument to `Url::link` method, because according to method specification (see `system/library/url.php` line 49) secure store URL won't be returned unless on explicitly request it. By default 3rd argument of `link` method is `false`.